src/main/java/net/minecraftforge/fml/relauncher/CoreModManager.java

```diff
@@ -412,7 +414,7 @@ public boolean accept(File pathname)
                 }
                 else
                 {
-                    FMLLog.log.trace("Found FMLCorePluginContainsFMLMod marker in {}, it will be examined later for regular @Mod instances",
+                    FMLLog.log.warn("Found FMLCorePluginContainsFMLMod marker in {}. This is not recommended, mods should be in an other jar as the coremod.",
```

Comment:
The text might be better this way:
`This is not recommended, @Mods should be in a separate jar from the coremod.`